### ISSUE TITLE
ai becomes braindead when in space

### ISSUE THREAD

**Issue by muburu** [space-wizards/space-station-14#28937]
## Description
if ai of any kind goes into space and is not on a tile, their ai takes a shit and they just stay there for all eternity unless moved onto a tile. also makes it super easy to cheese aggressive ai, since they won't try to attack either if stuck like this. makes space wreck mobs and dragon carp trivial to deal with if in space.

i don't know if this has been reported already (didn't see an open issue for it) or is just a limitation of the way pathfinding works, but i opened this issue since it's been a thing for as long as i can remember.

**Reproduction**
pretty much guaranteed to encounter this as space dragon, if carp try to follow you into space or wander into it, they just become stupid and don't do anything if they aren't on a tile.

**Screenshots**
![carp_stuck](https://github.com/space-wizards/space-station-14/assets/139181059/4c605412-f219-402f-8101-cc815959179a)
carp on the left and right are stuck even though they should easily be able to move that close to a tile. 
player controlled carp have free movement in space, so this is an issue exclusive to npcs.
(snip is from liltenhead dragon gameplay vid)

**Additional context**
pretty much happens with any npc that moves, even those that shouldn't be in space

**Comment by GroggleG** [comment]
pathing is gridbased so I mean shitcode central next stop

**Comment by Clonko** [comment]
Yeah, since pathing is gridbased the issue of AI not being able to go after players in space is also a issue.(They kinda just, forget you're there and go back to patroling.)
